new article on ordering of manifest elements

Rick-Kirkham opened the pull request:
Eventually we want to link to the new article from all the manifest reference topics, but that is out-of-scope for this PR.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/add-in-manifests.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/add-in-manifests.md) | :white_check_mark:Succeeded | |
[docs/develop/manifest-element-ordering.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/manifest-element-ordering.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C11%5C15%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201811150235108741-565%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=s4scFqV5n%2FrSxaDMm%2FmOf7XLSIPK7MM%2FAsXMSeWyKSc%3D&st=2018-11-15T02%3A31%3A32Z&se=2018-12-16T02%3A36%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+
+# How to find the proper order of manifest elements
+
+The XML elements in the manifest of an Office Add-in must not only be under to the proper parent element; they must be in a specific order, relative to each other, under the parent. For example, In the `<OfficeApp>` element, the `<Id>`, `<Version>`, `<ProviderName>` must appear in that order. If an `<AlternateId>` element is added, it must be between the `<ID>` and `<Version>` element. If any ele ...
```
"under to" -> "under"

AlexJerabek replied:
The second `<ID>` should be `<Id>`

AlexJerabek replied:
Consider changing "If any element is in the wrong order, your manifest will not be valid and your add-in will not load" to "Your manifest is not valid and your add-in will not load if any element is in the wrong order."

AlexJerabek commented on docs/develop/add-in-manifests.md:
```diff
@@ -29,6 +29,10 @@ An XML manifest file based on this schema enables an Office Add-in to do the fol
 The following table specifies the elements that are required for the three types of Office Add-ins.
 
+> [!NOTE]
+> There us also a mandatory order in which elements must appear within their parent element. For more information see [How to find the proper order of manifest elements](manifest-element-ordering.md).
```
General question: Should this article be called "The proper order of manifest elements" or even "Manifest element order"?

kbrandl replied:
"There us..." -> "There is..."

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+> [!NOTE]
+> The [Office Add-in Validator](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-manifest#validate-your-manifest-with-the-office-add-in-validator) uses the same error message when an element is out-of-order as it does when an element is under the wrong parent. The error will say that the child element is is not a valid child of the parent element. If you get such an error  ...
+
+The required ordering is specified in the XSD files in the [Schemas](https://github.com/OfficeDev/office-js-docs-pr/tree/docs/overview/schemas) folder. The XSD files are categorized into subfolders for taskpane, content, and mail add-ins.
```
This link 404s for me.

AlexJerabek replied:
I'd probably lead with this, instead of the example. That way, there's a point of reference as to why AlternativeId goes between Id and Version before that claim is made.

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+The XML elements in the manifest of an Office Add-in must not only be under to the proper parent element; they must be in a specific order, relative to each other, under the parent. For example, In the `<OfficeApp>` element, the `<Id>`, `<Version>`, `<ProviderName>` must appear in that order. If an `<AlternateId>` element is added, it must be between the `<ID>` and `<Version>` element. If any ele ...
+
+> [!NOTE]
+> The [Office Add-in Validator](https://docs.microsoft.com/en-us/office/dev/add-ins/testing/troubleshoot-manifest#validate-your-manifest-with-the-office-add-in-validator) uses the same error message when an element is out-of-order as it does when an element is under the wrong parent. The error will say that the child element is is not a valid child of the parent element. If you get such an error  ...
```
Please remove the "en-us" from this link. Also, should this be a relative link since it's in the same repo?

AlexJerabek replied:
Consider "The error will say that the" -> "The error says the".

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+
+The required ordering is specified in the XSD files in the [Schemas](https://github.com/OfficeDev/office-js-docs-pr/tree/docs/overview/schemas) folder. The XSD files are categorized into subfolders for taskpane, content, and mail add-ins.
+
+To find the correct order for the child elements of a given parent element, take the following steps. (This is a simplified process, but XSD files are quite complex. If these steps don't work, there are many resources on the Internet about reading XSD files.)
```
"but XSD" -> "as XSD" or "since XSD"

AlexJerabek replied:
Should we suggest a StackOverflow search or something for reading XSD files? Without a specific destination, I'd say it's better to omit a broad suggestion to look on the internet. We could also say something like "Fully parsing XSD files is out of the scope of this document."

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+
+To find the correct order for the child elements of a given parent element, take the following steps. (This is a simplified process, but XSD files are quite complex. If these steps don't work, there are many resources on the Internet about reading XSD files.)
+
+1. Open the subfolder under [Schemas](https://github.com/OfficeDev/office-js-docs-pr/tree/docs/overview/schemas) for the type of add-in that you are creating. 
```
Same 404 link comment.

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+To find the correct order for the child elements of a given parent element, take the following steps. (This is a simplified process, but XSD files are quite complex. If these steps don't work, there are many resources on the Internet about reading XSD files.)
+
+1. Open the subfolder under [Schemas](https://github.com/OfficeDev/office-js-docs-pr/tree/docs/overview/schemas) for the type of add-in that you are creating. 
+2. Open the XSD file where the parent element is defined as a complex type. (If you don't know which file has the definition, you may have to do step 3 on multiple files till you find it.)
```
Consider removing the parentheses.

AlexJerabek replied:
Consider "till" -> "until"

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
````diff
@@ -0,0 +1,50 @@
+  </xs:complexType>
+```
+
+The `<xs:sequence>` lists the possible child elements, *in the order in which they must appear*. This does *not* necessarily mean that all of them are mandatory. If the `minOccurs` value for a child element is **0**, then the child element is optional. *But if it is present, it must be in the order specified by the `<xs:sequence>` element*.
````
"This does *not* necessarily mean that all of them are mandatory." -> "This does *not* mean all of them are mandatory."

You could also get away with removing that second sentence and changing the third to "[…] then the child element is optional and can be omitted."

AlexJerabek replied:
As an entirely different alternative, you could define the `minOccurs` and `maxOccurs` attributes, mentioning what a min-0 implies.

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -0,0 +1,50 @@
+
+If there is no `<xs:sequence>` element, or there *is* but the child element is not listed (even though the reference documentation for the child element indicates that it *is* valid for the parent); then the parent element's complex type definition has been extended with additional child elements somewhere else in the XSD file. For example, the definition for the `OfficeApp` complex type does not ...
+
+To find the extended definitions follow these steps:
```
Nit: The following steps don't feel like steps, but rather a single step and a note. Maybe that's just my interpretation.

o365devx commented:
### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/manifest-element-ordering.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/manifest-element-ordering.md) | :warning:Warning | | [Details](#user-content-docs%2Fdevelop%2Fmanifest-element-ordering.md)
[docs/develop/add-in-manifests.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/add-in-manifests.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/toc.yml) | :white_check_mark:Succeeded | |

<a name="docs%2Fdevelop%2Fmanifest-element-ordering.md"></a>
### [docs/develop/manifest-element-ordering.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/manifest-element-ordering.md)
 - **Line 16**: **[Warning]** Invalid file link:(~/testing/troubleshoot-manifest#validate-your-manifest-with-the-office-add-in-validator).

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C11%5C15%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201811152309574632-565%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=51pI9x8%2Fm7ywxjW0knG2WrBWMsj5McEPwdSnhZYl%2B9w%3D&st=2018-11-15T23%3A06%3A18Z&se=2018-12-16T23%3A11%3A18Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

Rick-Kirkham commented:
@AlexJerabek I've addressed your feedback one way or another. I'm leaving the article title as is. I think it captures the problem that the article is trying to solve.

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/add-in-manifests.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/add-in-manifests.md) | :white_check_mark:Succeeded | |
[docs/develop/manifest-element-ordering.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/manifest-element-ordering.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C11%5C15%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201811152322433422-565%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=D1G0eIP2fD0XzyrokRWPhZgcii1PDSqLwK3huSMvrYo%3D&st=2018-11-15T23%3A19%3A02Z&se=2018-12-16T23%3A24%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

AlexJerabek commented on docs/develop/manifest-element-ordering.md:
```diff
@@ -6,17 +6,19 @@ ms.date: 11/16/2018
 # How to find the proper order of manifest elements
 
-The XML elements in the manifest of an Office Add-in must not only be under to the proper parent element; they must be in a specific order, relative to each other, under the parent. For example, In the `<OfficeApp>` element, the `<Id>`, `<Version>`, `<ProviderName>` must appear in that order. If an `<AlternateId>` element is added, it must be between the `<ID>` and `<Version>` element. If any ele ...
+The XML elements in the manifest of an Office Add-in must not only be under the proper parent element; they must be in a specific order, relative to each other, under the parent.
```
Nit: I keep getting tripped up by the "not". Consider:
"The XML elements in the manifest of an Office Add-in must be under the proper parent element **and** in a specific order, relative to each other, under the parent."

o365devx commented:
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/add-in-manifests.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/add-in-manifests.md) | :white_check_mark:Succeeded | |
[docs/develop/manifest-element-ordering.md](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/develop/manifest-element-ordering.md) | :white_check_mark:Succeeded | |
[docs/toc.yml](https://github.com/OfficeDev/office-js-docs-pr/blob/manifest-element-ordering/docs/toc.yml) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C11%5C16%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201811161948066424-565%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=47EXHfoflxzw7%2FDCsqvxDgwLUAU0ZsTMlifeMg1qAc4%3D&st=2018-11-16T19%3A44%3A32Z&se=2018-12-17T19%3A49%3A32Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.
